module/optee/voltd_regulator/src/mod_optee_voltd_regulator.c

```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright (c) 2023, Arm Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <fwk_assert.h>
+#include <fwk_log.h>
+#include <fwk_macros.h>
+#include <fwk_mm.h>
+#include <fwk_module.h>
+
+#include <mod_scmi_std.h>
+#include <mod_voltage_domain.h>
+#include <mod_optee_voltd_regulator.h>
+
+#include <stddef.h>
+
+#include <drivers/regulator.h>
+
+#define MOD_PREFIX "[optee-voltd-regulator] "
+
+/* Device context */
+struct optee_voltd_regulator_dev_ctx {
+    struct regulator *regulator;
+    bool enabled;
+};
+
+/* Module context */
+struct optee_voltd_regulator_ctx {
+    struct optee_voltd_regulator_dev_ctx *dev_ctx_table;
+    unsigned int dev_count;
+};
+
+/* A single instance handles all voltage regulators abstracted by regulator.h */
+static struct optee_voltd_regulator_ctx module_ctx;
+
+static struct optee_voltd_regulator_dev_ctx *find_ctx(fwk_id_t dev_id)
```

Comment:
`get_ctx` instead? since it a search is not being perfomed

Reply:
fair